ush/wave_extractvars.sh

```diff
@@ -0,0 +1,34 @@
+#! /usr/bin/env bash                                                                                                                                                                          
+
+################################################################################
+## UNIX Script Documentation Block
+## Script name:         wave_extractvars.sh
+## Script description:  Extracts variables from wave products
+##                      and saves these variables in arcdir
+#######################
+# Main body starts here
+#######################
+
+source "${USHgfs}/preamble.sh"
+
+subdata=${1}
+
+[[ -d "${subdata}" ]] || mkdir -p "${subdata}"
+
+for (( nh = FHOUT_WAV_EXTRACT; nh <= FHMAX_WAV; nh = nh + FHOUT_WAV_EXTRACT )); do
+  fnh=$(printf "%3.3d" "${nh}")
+
+  infile=${COMIN_WAVE_GRID}/${RUN}wave.t${cyc}z.global.${wavres}.f${fnh}.grib2
+  outfile=${subdata}/${RUN}wave.t${cyc}z.global.${wavres}.f${fnh}.grib2
+  rm -f "${outfile}" #remove outfile if it already exists before extraction
+
+  if [[ -f "${infile}" ]]; then #check if input file exists before extraction
+    # shellcheck disable=SC2312 
+    ${WGRIB2} "${infile}" | grep -F -f "${varlist_wav}" | ${WGRIB2} -i "${infile}" -append -grib "${outfile}">/dev/null
```

Comment:
Should we be checking the status of this call? The output is being sent to `/dev/null`.
If there is any issue with this call, it will not be seen.